### PR TITLE
Vjogi/ravcom 444 upgrade core module

### PR DESCRIPTION
**Pull request by vijayjogi-dm**
Rel package related changes.

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java [hunk 1]
```diff
@@ -144,14 +153,9 @@ public class RelToSqlConverter extends SqlImplementor
   public RelToSqlConverter(SqlDialect dialect) {
     super(dialect, DEFAULT_BLOAT);
     style = new QueryStyle();
-    dispatcher = ReflectUtil.createMethodDispatcher(Result.class, this, "visit",
-      RelNode.class);
-  }
-  public RelToSqlConverter(SqlDialect dialect, QueryStyle style) {
```
Comment (Mahesh-Raut) [inline core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java:150]: Please revert back this Constructor.

File: core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java [hunk 2]
```diff
@@ -1349,7 +1643,7 @@ public List<SqlNode> createAsFullOperands(RelDataType rowType, SqlNode leftOpera
     selectList.add(node);
   }
 
-  protected void parseCorrelTable(RelNode relNode, Result x) {
+  private void parseCorrelTable(RelNode relNode, Result x) {
```
Comment (Mahesh-Raut) [inline core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java:1646]: Please make it as protected as this method is used in child classes.

Comment (vijayjogi-dm) [reply core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java:1646]: I will address suggested changes in next PR

File: core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java (hunk 1 above)
Comment (vijayjogi-dm) [reply core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java:150]: I will address suggested changes in next PR

**Pull request merged**